camelot/core/files/storage.py

```diff
@@ -99,154 +110,176 @@ def __init__( self, upload_to = '',
         #
 
     @property
-    def upload_to(self):
-        if self._upload_to == None:
+    def upload_to(self) -> PurePosixPath:
+        """Return the directory path to upload files"""
+        if self._upload_to is None:
             root = self._root or settings.CAMELOT_MEDIA_ROOT
-            import os
-            if callable( root ):
+            if callable(root):
                 root = root()
-            self._upload_to = os.path.join( root, self._subfolder )
+            root = PurePosixPath(root)
+            self._upload_to = root.joinpath(self._subfolder)
         return self._upload_to
-        
-    def available(self):
-        """Verify if the storage is available
 
+    def available(self) -> bool:
+        """
+        Verify if the storage is available, or create if it doesn't exist
         :return: True if the storage is available, False otherwise
         """
-        import os
         try:
-            if not os.path.exists( self.upload_to ):
-                os.makedirs( self.upload_to )
+            if not (p := Path(self.upload_to)).exists():
+                p.mkdir(parents=True)
             return True
         except Exception as e:
-            logger.warn( 'Could not access or create path %s, files will be unreachable' % self.upload_to, exc_info = e )
+            logger.warning(f'Could not access or create path {self.upload_to}, files will be unreachable', exc_info=e)
+            return False
 
     def writeable(self):
-        """Verify if the storage is available and writeable
+        """Verify if the storage is available and writable
 
-        :return: True if the storage is writeable, False otherwise
+        :return: True if the storage is writable, False otherwise
         """
-        import os
         if self.available():
-            return os.access(self.upload_to, os.W_OK) 
-        
-    def exists( self, name ):
-        """True if a file exists given some name"""
-        if self.available():
-            import os
-            return os.path.exists( self.path( name ) )
+            return os.access(Path(self.upload_to), os.W_OK)
         return False
-        
-    def list(self, prefix='*', suffix='*'):
-        """Lists all files with a given prefix and or suffix available in this storage
 
-        :return: a iterator of StoredFile objects
+    @staticmethod
+    def exists(name: PathType) -> bool:
+        """Check if a file exists given its name
+
+        :param name: Name of the file
+        :return: True if the file exists, False otherwise
+        """
+        return Path(name).exists()
```

Comment:
di nt juist heh, die gaat checken of het relatief pad bestaat, hier best een test van maken, ik denk dat deze bij in de 'list' unittest kan

Reply:
kan ook onmogelik een static method zijn ...

Reply:
fixed